os/mm/mm_heap/mm_mallinfo.c

```diff
@@ -111,9 +113,31 @@ int mm_mallinfo(FAR struct mm_heap_s *heap, FAR struct mallinfo *info)
 
 		mm_takesemaphore(heap);
 
-		for (node = heap->mm_heapstart[region]; node < heap->mm_heapend[region]; node = (struct mm_allocnode_s *)((char *)node + node->size)) {
+		prev = NULL;
+		node = heap->mm_heapstart[region];
+		next = (struct mm_allocnode_s *)((char *)node + node->size);
+
+		for (; node < heap->mm_heapend[region]; prev = node, node = next, next = (struct mm_allocnode_s *)((char *)next + next->size)) {
 			mvdbg("region=%d node=%p size=%p preceding=%p (%c)\n", region, node, node->size, (node->preceding & ~MM_ALLOC_BIT), (node->preceding & MM_ALLOC_BIT) ? 'A' : 'F');
 
+			if (node->size != (next->preceding & ~MM_ALLOC_BIT)) {
+				mdbg("ERROR: Heap node corruption detected in one of following nodes\n");
+				if (prev) {
+					mdbg("Previous node addr = 0x%08x size = %u preceding size = %u\n", prev, prev->size, prev->preceding & ~MM_ALLOC_BIT);
+				}
+				mdbg("Current  node addr = 0x%08x size = %u preceding size = %u\n", node, node->size, node->preceding & ~MM_ALLOC_BIT);
```

Comment:
mdbg display function name as prefix eh?
If I were you I would take below style.
```
#ifdef CONFIG_DEBUG_MM_ERROR
 dbg("Current  node addr = 0x%08x size = %u preceding size = %u\n", node, node->size, node->preceding & ~MM_ALLOC_BIT);
 dbg("Next     node addr = 0x%08x size = %u preceding size = %u\n", next, next->size, next->preceding & ~MM_ALLOC_BIT);
#endif
```

Reply:
But mdbg internally uses dbg.. and it prints the function name prefix.

Reply:
```c
#ifdef CONFIG_DEBUG_MM_ERROR
dbg(....);
#endif
```
and
```c
mdbg(....);
```
are exactly the same.